src/engine/source/builder/builders/opBuilderHelperMap.cpp

```diff
@@ -485,6 +486,91 @@ base::Expression opBuilderHelperStringConcat(const std::any& definition)
+    helper::base::checkParameterType(parameters[0], helper::base::Parameter::Type::VALUE);
+    const auto separator = parameters.at(0);
+
+    // Check Array reference parameter
+    helper::base::checkParameterType(parameters[1],
+                                     helper::base::Parameter::Type::REFERENCE);
+    const auto arrayName = parameters.at(1);
+
+    name = helper::base::formatHelperFilterName(name, targetField, parameters);
+
+    // Tracing
+    const auto successTrace = fmt::format("[{}] -> Success", name);
+    const auto failureTrace1 =
+        fmt::format("[{}] -> Failure: [{}] param should be a string", name, targetField);
+    const auto failureTrace2 = fmt::format(
+        "[{}] -> Failure: parameter should be a non empty string array", name);
+    const auto failureTrace3 =
+        fmt::format("[{}] -> Failure: array parameter with invalid json path", name);
+
+    // Return Term
+    return base::Term<base::EngineOp>::create(
+        name,
+        [=,
+         targetField = std::move(targetField),
+         separator = std::move(separator),
+         arrayName = std::move(arrayName)](
+            base::Event event) -> base::result::Result<base::Event>
+        {
+            // Getting separator field name
+            const std::string resolvedSeparator = separator.m_value;
+
+            // Getting array field, must be a reference
+            std::optional<std::vector<json::Json>> stringJsonArray;
+            try
+            {
+                stringJsonArray = event->getArray(arrayName.m_value);
+                if (!stringJsonArray.has_value() || stringJsonArray.value().size() == 0)
+                {
+                    return base::result::makeFailure(event, failureTrace2);
+                }
+            }
+            catch (const std::runtime_error& e)
+            {
+                return base::result::makeFailure(event, failureTrace3);
+            }
+
+            // TODO: avoid dumping JsonArray to a string one
+            std::vector<std::string> stringArray;
+            for (const auto& s_param : stringJsonArray.value())
+            {
+                if (s_param.isString())
+                {
+                    stringArray.emplace_back(s_param.getString().value());
+                }
+                else
+                {
+                    return base::result::makeFailure(event, failureTrace1);
+                }
+            }
+
+            // lambda expression for accumulate concatenation
+            auto concatPlusSeparator = [&](std::string a, std::string b) {
+                return std::move(a) + resolvedSeparator + std::move(b);
+            };
+
+            // accumulated concation without trailing indexes
+            std::string composedValueString =
+                std::accumulate(std::next(stringArray.begin()),
+                                stringArray.end(),
+                                stringArray[0],
+                                concatPlusSeparator);
```

Comment:
Maybe avoid excessive copying with something like this?
```
            std::vector<std::string> stringArray;
            ssize_t resultSize = 1; // endline "\0"
            std::string result {};
            for (const auto& s_param : stringJsonArray.value())
            {
                if (s_param.isString())
                {
                    auto strVal = s_param.getString().value();
                    resultSize += strVal.size() + resolvedSeparator.size();
                    stringArray.emplace_back(std::move(resultSize));
                }
                else
                {
                    return base::result::makeFailure(event, failureTrace1);
                }
            }

            result.reserve(resultSize);

            auto first = false;
            for (const auto& item : stringArray)
            {
                result.append(first ? "" : resolvedSeparator);
                result.append(item);
            }
```

Reply:
ok